src/valor_lite/cache.py

```diff
@@ -0,0 +1,477 @@
+class DataType(StrEnum):
+    INTEGER = "int"
+    FLOAT = "float"
+    STRING = "string"
+    TIMESTAMP = "timestamp"
+
+    def to_py(self):
+        """Get python type."""
+        match self:
+            case DataType.INTEGER:
+                return int
+            case DataType.FLOAT:
+                return float
+            case DataType.STRING:
+                return str
+            case DataType.TIMESTAMP:
+                return datetime
+
+    def to_arrow(self):
+        """Get arrow type."""
+        match self:
+            case DataType.INTEGER:
+                return pa.int64()
+            case DataType.FLOAT:
+                return pa.float64()
+            case DataType.STRING:
+                return pa.string()
+            case DataType.TIMESTAMP:
+                return pa.timestamp("us")
+
+
+def convert_type_mapping_to_schema(
+    type_mapping: dict[str, DataType] | None
+) -> list[tuple[str, pl.DataType]]:
+    """
+    Convert type mapping to a pyarrow schema input.
+
+    Parameters
+    ----------
+    type_mapping : dict[str, DataType] | None
+        A map from string key to datatype. Treats input of `None` as empty mapping.
+
+    Returns
+    -------
+    list[tuple[str, pyarrow.lib.DataType]]
+        A list of field name, field type pairs that can be used as input to pyarrow.schema.
+    """
+    if not type_mapping:
+        return []
+    return [(k, DataType(v).to_arrow()) for k, v in type_mapping.items()]
+
+
+class CacheFiles:
+    def __init__(self, path: str | Path):
+        self._path = Path(path)
+
+    @property
+    def path(self) -> Path:
+        """Cache directory path."""
+        return self._path
+
+    @staticmethod
+    def _generate_config_path(path: str | Path) -> Path:
+        """
+        Generate cache configuration path.
+
+        Parameters
+        ----------
+        path : str | Path
+            Where the cache is stored.
+
+        Returns
+        -------
+        Path
+            Configuration filepath.
+        """
+        return Path(path) / ".cfg"
+
+    def get_files(self) -> list[Path]:
+        """
+        Retrieve all files.
+
+        Returns
+        -------
+        list[Path]
+            A list of paths to files in the cache.
+        """
+        if not self.path.exists():
+            return []
+        files = []
+        for entry in os.listdir(self._path):
+            full_path = os.path.join(self._path, entry)
+            if os.path.isfile(full_path):
+                files.append(Path(full_path))
+        return files
+
+    def get_dataset_files(self) -> list[Path]:
+        """
+        Retrieve all dataset files.
+
+        Returns
+        -------
+        list[Path]
+            A list of paths to dataset files in the cache.
+        """
+        if not self.path.exists():
+            return []
+        return [
+            Path(filepath) for filepath in glob.glob(f"{self._path}/*.parquet")
+        ]
+
+
+class CacheReader(CacheFiles):
```

Comment:
When we add memory support can this be something like CacheFiles | None? Or is this a subclassing type thing, in which case i am not sure how we add memory support 🤔